src/plugin/ipc/file/filewrappers.cpp

```diff
@@ -371,7 +371,13 @@ static int _open_open64_work(int(*fn) (const char *path, int flags, ...),
     newpath = currPtsDevName;
   }
 
-  int fd = (*fn) (newpath, flags, mode);
+  dmtcp::string phys_path_string = "";
+  const char *phys_path =  virtual_to_physical_path ?
+                           virtual_to_physical_path(newpath, phys_path_string):
+                           newpath;
+
```

Comment:
@mossberg: Concerning your interface proposal in the "conversation" tab, I do like your idea about using a macro.  These three lines get repeated often, and should be replaced by a macro.  How about the following?
```
// Note that virtual_to_physical_path is a weak symbol. 
// (It is a NULL ptr if not linked to an object defining this.)
#define VIRTUAL_TO_PHYSICAL_PATH(virt_path, phys_path_string) \
                  (virtual_to_physical_path ? \
                   virtual_to_physical_path(newpath, phys_path_string) : \
                   newpath)
```
Then, the line above becomes:
```
const char *phys_path = VIRTUAL_TO_PHYSICAL_PATH(newpath,
                                                 phys_path_string)
```
and we can invoke the macro in a similar way on the lines below.

(Note also that I try to keep the code within 80 columns.  And I use a long macro name for the macro (no abbreviations), since the macro name is not defined locally, and otherwise the user has to search for the exact name if one of several possible abbreviations is chosen.)

And note that I added to your line note where you diagnosed the bug concerning `JASSERT` and `phys_path_string`.  I'm proposing a bug fix there (and maybe it's the same one that you already had in mind).